### DISCUSSION TITLE
Syntax help for a Suggester for VDATE (with Natrual Language Date) input in Quick Add capture format

### DISCUSSION THREAD

**Discussion by guycalledxan** [chhoumann/quickadd#690] [Q&A]
# What I’m trying to do

I use Quick Add’s capture format to create tasks. I’m trying to get an optional Reminder date added to a task that, when chosen, will use a Suggester to populate the date. I can do the former, be not the latter.

Basically, I got annoyed with having to type “Tomorrow” or “Next Week” and figured I could click on suggester options instead.

I’ve experimented with a few things but can’t crack it.

Here’s the full Capture format:

`#{{FIELD:tags}} :pencil2:{{selected}}{{VALUE:, <% tp.system.clipboard()%>}} {{VALUE:Task Description}} _-{{VALUE:, Reminder: (@{{VDATE:Reminder-Date, [[YYYY-MM-DD]] HH:mm}})}} :heavy_plus_sign: [[{{Date}}]] {{VALUE::stop_button:, , :small_red_triangle:, :arrow_double_up:, :arrow_up_small:, :arrow_down_small:, :arrow_double_down:}}`

# Things I have tried

As above, the closest I can get so far is:

`{{VALUE:, Reminder: (@{{VDATE:Reminder-Date, [[YYYY-MM-DD]] HH:mm}})}}`

That gives me an optional Reminder date Reminder: (@ token and if I select the blank option I don’t get any other text there, which is good.

In this one, the VDATE resolves as Reminder: (@{{VDATE:monday)}}:
`{{VALUE:, Reminder: (@{{VDATE:{{FIELD:reminder_time_options}}, [[YYYY-MM-DD]] HH:mm}})}} :heavy_plus_sign: [[{{Date}}]] {{VALUE::stop_button:, , :small_red_triangle:, :arrow_double_up:, :arrow_up_small:, :arrow_down_small:, :arrow_double_down:}}`

Any use of commas in VDATE causes it to behave like a dropdown but won’t resolve as a date:
`{{VALUE:, Reminder: (@{{VDATE:{{VALUE:monday, tomorrow}}, [[YYYY-MM-DD]] HH:mm}})}}`

`{{VALUE:, Reminder: (@{{VDATE:monday, tomorrow, [[YYYY-MM-DD]] HH:mm}})}}`

(This is a dupe of a [post](https://forum.obsidian.md/t/syntax-help-for-a-suggester-for-vdate-natrual-language-date-input-in-quick-add-capture-format/82345) on the Obsidian forums. That got no reply so I thought I'd try here. Hope that's OK)

**Comment by chhoumann** [comment]
Sorry for the very slow reply.

First, why the nested-token attempts can't work: inside `{{VDATE:name, format}}` the comma is the separator between the variable name and the date format, and QuickAdd doesn't expand other tokens inside a VDATE. So `{{VDATE:{{VALUE:...}}, ...}}` can never parse - nothing resolves the inner token before VDATE reads it.

The good news is VDATE has grown since 2024 and probably covers most of what you were reaching for:

- It parses natural language as you type, with a live preview: "tomorrow", "next friday", "in two weeks".
- Short aliases: `t` = today, `tm` = tomorrow, `yd` = yesterday (configurable in settings).
- Natural-language defaults: `{{VDATE:reminder,YYYY-MM-DD HH:mm|tomorrow}}` pre-fills tomorrow.
- `|time` adds a time picker; `|optional` lets you skip the prompt.

Typing `tm` is two keystrokes, so a suggester may be more clicks than it saves - but if you specifically want a clickable list, here's the macro version:

1. User script, e.g. `scripts/pick-reminder.js`:

```js
module.exports = async (params) => {
	const { quickAddApi } = params;
	const moment = window.moment;
	const options = {
		"Tomorrow": () => moment().add(1, "day"),
		"Next week": () => moment().add(1, "week"),
		"Monday": () => moment().day(8),
		"No reminder": () => null,
	};
	const labels = Object.keys(options);
	const pick = await quickAddApi.suggester(labels, labels);
	if (pick === undefined) {
		params.abort("Cancelled.");
		return;
	}
	const when = options[pick]();
	params.variables.reminder = when ? when.format("YYYY-MM-DD HH:mm") : "";
};
```

2. Make a Macro ("Reminder") with two steps: the User Script above, then your Capture.

3. In the Capture, reference `{{VALUE:reminder}}` wherever you want the date. Because the script pre-fills it, the capture doesn't prompt for it again. (Adjust the option list to taste - each entry just returns a moment date.)

One more thing from your format string: `<% tp.system.clipboard() %>` is Templater syntax and won't run inside a QuickAdd format - use QuickAdd's own `{{CLIPBOARD}}` token instead. It pastes the clipboard text, and if the clipboard holds an image rather than text, it saves the image and inserts an embed.

Picking "Tomorrow" wrote `- [ ] Follow up (reminder: 2026-07-07 19:27) - https://example.com/article` - the date came from the click, the URL from `{{CLIPBOARD}}`, and the reminder was not prompted a second time.

Docs: https://quickadd.obsidian.guide/docs/FormatSyntax#vdate
